src/config.ts

```diff
@@ -39,3 +57,17 @@ function loadRequiredEnvVar(envVarName: string): string {
 
   return envVar;
 }
+
+function mapEnvironmentModeFromStringToEnum(
+  environmentMode: string,
+): EnvironmentMode {
+  if (environmentMode === "staging") {
+    return EnvironmentMode.Staging;
+  }
+
+  if (environmentMode === "production") {
+    return EnvironmentMode.Production;
+  }
+
+  return EnvironmentMode.Local;
```

Comment:
Could you do something like this instead to avoid hardcoding the if statements?
```suggestion
  return EnvironmentMode[environmentMode] ?? EnvironmentMode.Local;
```

Reply:
Unfortunately, I could not find a way to make this work 😕 

Here is the Typescript error:
```
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'typeof EnvironmentMode'.
  No index signature with a parameter of type 'string' was found on type 'typeof EnvironmentMode'.
```
I can try to spend time looking for a better solution.

Reply:
Totally fair - what you have here will work and it's not like we're going to continually be adding environments.  I was more curious since I've used that type of pattern in other languages and naively assumed it would work here 😅

Reply:
Looks like there is a way to make it work but it will be pretty much equivalent in terms of added code.
https://futurestud.io/tutorials/typescript-using-a-string-as-enum-key#:~:text=and%20Omit%20Methods-,Using%20a%20String%20to%20Access%20Enum%20Values%20in%20TypeScript,t%20use%20a%20string%20value.

Reply:
Gotcha - appreciate you digging into it